src/ncp/ncp_base.cpp

```diff
@@ -2524,17 +2566,33 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_BEACON:
-            mShouldSignalEndOfScan = false;
             errorCode = otActiveScan(
                             mChannelMask,
                             mScanPeriod,
                             &HandleActiveScanResult_Jump,
                             this
                         );
+
+            if (errorCode == kThreadError_None)
+            {
+                mShouldSignalEndOfScan = false;
```

Comment:
This is addressing a (rare) corner case where the host (wpantund) has issued a new scan command and the we have not yet sent the event signaling the end of previous scan (no buffer space to send). In this case, the host has likely given up (timed out) on the previous scan and if we send a "end of scan" signal/event it can be confused as the end of current/new scan.  The "end of scan" message uses TID 0 (which is for sending async events) so there is no way for host/wpantund to associate it with a specific scan command. 

Note that we clear this flag only if the new scan command starts successfully.